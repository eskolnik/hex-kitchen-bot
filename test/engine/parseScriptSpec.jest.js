// /**
//  * Tests for the script parser
//  */
const { parseScript } = require("../../src/engine/parseScript");
const fs = require("fs");
const mockScriptPath = "test/engine/mockScript.json";

describe("parseScript", () => {
    test("returns a set of game objects based on the input JSON", () => {
        const scriptFile = fs.readFileSync(mockScriptPath);

        const script = parseScript(scriptFile);

        expect(Object.keys(script.events).length).toEqual(16);
        expect(script.events["5"]).toMatchSnapshot();
        expect(script.channels.length).toEqual(4);
        expect(script.categoryChannels.length).toEqual(2);
    });
});
