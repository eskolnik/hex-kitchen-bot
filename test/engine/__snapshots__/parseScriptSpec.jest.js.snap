// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseScript returns a set of game objects based on the input JSON 1`] = `
Object {
  "content": "ðŸšª",
  "eventsTriggered": Array [
    "6",
    "7",
    "8",
    "9",
    "10",
  ],
  "id": 5,
  "messages": Array [],
  "options": Object {
    "autoRemove": true,
    "max": 1,
  },
  "status": "unavailable",
  "targetEvent": 4,
  "type": "EmojiReact",
}
`;
